### PR TITLE
RHIROS-1220 Kruize as ClowdApp kind

### PR DESCRIPTION
**Pull request by saltgen**
### Changes
Splitting kruize deployment into a new ClowdApp

Steps to verify on EE:

### 2. Update local bonfire config as below.

Path: ```~/<user>/.config/bonfire/config.yaml```

```
# Bonfire deployment configuration

# Defines where to fetch the file that defines application configs
appsFile:
  host: gitlab
  repo: insights-platform/cicd-common
  path: bonfire_configs/ephemeral_apps.yaml

# (optional) define any apps locally. An app defined here with <name> will override config for app
# <name> in above fetched config.
apps:
- name: kruize
  components:
    - name: kruize
      host: github
      repo: RedHatInsights/ros-ocp-backend
      path: /kruize-clowdapp.yaml
```

### 3. Deploy updated ros-ocp-backend clowder changes

```
oc process --local -p ENV_NAME=env-<reserved EE namespace> -p IMAGE_TAG=pr-102-c890c67 -p RBAC_ENABLE=false -f ~/<path>/ros-ocp-backend/clowdapp.yaml | oc apply -n <reserved EE namespace> -f -
```

### 4. Deploy new kruize clowder changes

```
oc process --local -p ENV_NAME=env-<reserved EE namespace> -p KRUIZE_IMAGE_TAG=190a6cc -f ~/<path>/ros-ocp-backend/kruize-clowdapp.yaml | oc apply -n <reserved EE namespace> -f -
```

### 5. Generate test recommendation

```make archive-to-minio env=<reserved EE namespace>```

### 6. Update Makefile

Add query param ```?start_date=2023-04-01``` to Makefile L#121

### 7. Request recommendation from API

```make get-recommendations env=<reserved EE namespace>```

### Note:
Kruize hostname has changed and  it has its own database called ```kruize``` inside kruize-db pod, even in EE.

```
kruize-db-764c5dd5ff-8b4n5                                       1/1     Running   1 (43m ago)   43m
kruize-recommendations-6c847ffdb8-r575m                          1/1     Running   0             43m
puptoo-processor-67ff5fc6bb-ngjq4                                1/1     Running   0             48m
puptoo-redis-68c8b74bff-glqdg                                    1/1     Running   0             48m
rbac-db-784c5c7c68-85dkg                                         1/1     Running   1 (47m ago)   47m
rbac-redis-54c955d995-rvw24                                      1/1     Running   0             47m
rbac-service-688f9bc9c6-zv98z                                    2/2     Running   0             47m
ros-ocp-backend-api-84b77b9696-lwf9z                             2/2     Running   0             17m
ros-ocp-backend-db-5d4bc5bfcf-xbn8x                              1/1     Running   1 (47m ago)   47m
ros-ocp-backend-processor-7fdc86b9c4-mlsrk                       1/1     Running   0             24m
ros-ocp-backend-recommender-76f557f86b-tzn5t                     1/1     Running   0             24m

```

### CODE REVIEW

File: kruize-clowdapp.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            value: "false"
+          - name: hibernate_timezone
+            value: "UTC"
+          - name: SSL_CERT_DIR
+            value: ${SSL_CERT_DIR}
+    database:
+      name: kruize
+      version: 13
+    testing:
+      iqePlugin: ros-ocp
```
Comment (patilsuraj767) [inline kruize-clowdapp.yaml:82]: I don't think we have any iqe tests specific to kruize so this should be removed.

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]
On [line 14](https://github.com/RedHatInsights/ros-ocp-backend/blob/main/clowdapp.yaml#L14) in clowdapp file under `dependencies` we should add `kruize` as a dependencies of ros-ocp.

**Comment by saltgen** [comment]
The ci.int failure is cause of the new dependency `kruize`, guess we need to update the deploy command there?

**Comment by patilsuraj767** [comment]
> The ci.int failure is cause of the dependency `kruize`, guess we need to update the deploy command there?

Yeah. so thought regarding it again. All though the kruize is a separate application and has a separate clowdapp fill still it the part of ROS. For app-interface ros-ocp and kruize we still be a single app which means running the `bonfire deploy ros` will also deploy the kruize hence no need to add it as dependency. It can be removed.

**Comment by saltgen** [comment]
> > The ci.int failure is cause of the dependency `kruize`, guess we need to update the deploy command there?
> 
> Yeah. so I thought regarding it again. All though the kruize is a separate application and has a separate clowdapp file still it the part of ROS. For app-interface ros-ocp and kruize will still be a single app which means running the `bonfire deploy ros` will also deploy the kruize hence no need to add it as dependency. It can be removed.

Change pushed

**Review by patilsuraj767 — APPROVED** [review]
LGTM 👍 Thanks @saltgen

**Pull request merged**